snuba/clickhouse/formatter/expression.py

```diff
@@ -212,16 +217,7 @@ def _format_string_literal(self, exp: Literal) -> str:
         return "$S"
 
     def _format_number_literal(self, exp: Literal) -> str:
-        return "$N"
-
-    def _format_boolean_literal(self, exp: Literal) -> str:
-        return "$B"
-
-    def _format_datetime_literal(self, exp: Literal) -> str:
-        return "$DT"
-
-    def _format_date_literal(self, exp: Literal) -> str:
-        return "$D"
+        return str(_gen_random_number())
```

Comment:
What about hardcoding 0 or -1 ? Basically something that tells us for sure that this is a value we tweaked.
A random number could mislead people in thinking that it is the real value provided by the user.